catroid/src/main/java/org/catrobat/catroid/stage/StageListener.java

```diff
@@ -564,6 +561,18 @@ public void render() {
 			screenshot = ScreenUtils
 					.getFrameBufferPixels(screenshotX, screenshotY, screenshotWidth, screenshotHeight, true);
 			makeScreenshot = false;
+			if (screenshot == null || screenshotName == null || screenshotName.matches("\\s*")) {
```

Comment:
Please move these lines to an extra Method  (makeScreenshot or something similar) and instead of creating a new callback method you could pass the given one.